cpp/test/refinement/interval.cpp

```diff
@@ -0,0 +1,290 @@
+    CHECK(x[0] == 0.0);
+    CHECK(x[1] == 0.0);
+    CHECK(x[2] == 0.0);
+
+    CHECK(x[3] == 0.5);
+    CHECK(x[4] == 1.0);
+    CHECK(x[5] == 2.0);
+
+    CHECK(x[6] == 0.75);
+    CHECK(x[7] == 1.5);
+    CHECK(x[8] == 3.0);
+
+    CHECK(x[9] == 1.0);
+    CHECK(x[10] == 2.0);
+    CHECK(x[11] == 4.0);
+  }
+
+  // Check topology
+  {
+    auto topology = refined_mesh.topology_mutable();
+    CHECK(topology->dim() == 1);
+
+    topology->create_connectivity(0, 1);
+    auto v_to_e = topology->connectivity(0, 1);
+
+    CHECK(v_to_e->num_links(0) == 1);
+    CHECK(v_to_e->links(0)[0] == 0);
+
+    CHECK(v_to_e->num_links(1) == 2);
+    CHECK(v_to_e->links(1)[0] == 0);
+    CHECK(v_to_e->links(1)[1] == 1);
+
+    CHECK(v_to_e->num_links(2) == 2);
+    CHECK(v_to_e->links(2)[0] == 1);
+    CHECK(v_to_e->links(2)[1] == 2);
+
+    CHECK(v_to_e->num_links(3) == 1);
+    CHECK(v_to_e->links(3)[0] == 2);
+  }
+
+  // Check parent edges
+  {
+    CHECK(parent_edge.size() == 3);
+
+    CHECK(parent_edge[0] == 0);
+    CHECK(parent_edge[1] == 1);
+    CHECK(parent_edge[2] == 1);
+  }
+}
+
+TEST_CASE("Interval Refinement (parallel)", "refinement,interval,paralle")
+{
+  /**
+  Produces an interval with communicator size intervals. Every process is
+  assigned one intervall and we refine uniformly.
+  */
+
+  const auto comm_size = dolfinx::MPI::size(MPI_COMM_WORLD);
+  const auto rank = dolfinx::MPI::rank(MPI_COMM_WORLD);
+
+  if (comm_size == 1)
+    return;
+
+  auto create_mesh = [&]()
+  {
+    std::vector<double> x;
+    std::vector<std::int64_t> cells;
+    fem::CoordinateElement<double> element(mesh::CellType::interval, 1);
+    if (rank == 0)
+    {
+      for (std::int64_t i = 0; i < comm_size + 1; i++)
+        x.insert(x.end(), {static_cast<double>(i) / comm_size,
+                           static_cast<double>(i) + 1, 2. * i + comm_size});
+      for (std::int64_t i = 0; i < 2 * comm_size; i++)
+      {
+        auto div = std::div(i, static_cast<std::int64_t>(2));
+        cells.push_back(div.quot + div.rem);
+      }
+    }
+
+    auto partitioner
+        = [](MPI_Comm /* comm */, int /* nparts */,
+             const std::vector<mesh::CellType>& /* cell_types */,
+             const std::vector<std::span<const std::int64_t>>& /* cells */)
+        -> graph::AdjacencyList<std::int32_t>
+    {
+      return graph::AdjacencyList<std::int32_t>(
+          dolfinx::MPI::size(MPI_COMM_WORLD));
+    };
+
+    auto commt = rank == 0 ? MPI_COMM_SELF : MPI_COMM_NULL;
+    return mesh::create_mesh(MPI_COMM_WORLD, commt, cells, element, commt, x,
+                             {x.size() / 3, 3}, partitioner);
+  };
+
+  mesh::Mesh<double> mesh = create_mesh();
+  mesh.topology()->create_connectivity(1, 0);
+
+  // complete refinement
+  {
+    auto [refined_mesh, parent_edges]
+        = refinement::refine_interval(mesh, std::nullopt, false);
+
+    // Check geometry
+    {
+
+      auto x = refined_mesh.geometry().x();
```

Comment:
https://github.com/catchorg/Catch2/blob/devel/docs/comparing-floating-point-numbers.md

Never compare using equality of floating point types.